### PR TITLE
Feat/customer profile update

### PR DESCRIPTION
**Pull request by hemal-digiflux**

### CODE REVIEW

File: src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                </tr>,
+            );
+        });
+        return rows;
+    };
+
+    render() {
+        const { customerUser } = this.props;
+        return (
+            <Container fluid>
```
Comment (Diego-Romero) [inline src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx]: This also needs to be hooked to the stub server and have form validation

File: src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+interface PropsFromState {
+    customerUser: customerUserInterface;
+    customerDashboard: customerDashboardInterface;
+}
+
+type CustomerDashboardPageAllProps = RouteComponentProps & ConnectedReduxProps & PropsFromState;
+
+export class CustomerDashboard extends React.Component<CustomerDashboardPageAllProps> {
+    renderUpcomingBookings = () => {
```
[the remark this answers is not in the document]
Comment (hemal-digiflux) [reply src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx]: I will add this in Customer Dashboard code, this is actually not used inside this page, it was by mistake. So removing this function for CustomerProfile Page

File: src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import CONFIG from '../../config/config';
+
+interface PropsFromState {
+    customerUser: customerUserInterface;
+    customerDashboard: customerDashboardInterface;
+}
+
+type CustomerDashboardPageAllProps = RouteComponentProps & ConnectedReduxProps & PropsFromState;
+
+export class CustomerDashboard extends React.Component<CustomerDashboardPageAllProps> {
```
[the remark this answers is not in the document]
Comment (hemal-digiflux) [reply src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx]: Remove Extra code

File: src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx (hunk 1 above)
Comment (hemal-digiflux) [reply src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx]: This is coming from the customerUser object already hooked from the stub server, but I think now, we have to handle the refresh browser scenario as well, for which we have created a ticket.

Comment (hemal-digiflux) [reply src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx]: We will have Formik + Yup here.

Comment (Diego-Romero) [reply src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx]: What is stoping us from introducing it now? Sorry I don't understand

Comment (Diego-Romero) [reply src/ecosystems/CustomerProfileUpdate/CustomerProfileUpdate.page.tsx]: So this page needs to load the user when the page loads on componentDidMount, and it should also send a PUT to the server with the form information - once this is valid.

File: src/redux/api.ts [hunk 4]
```diff
@@ -31,14 +34,14 @@ export async function apiLoginCustomerUser({
...
-export async function apiGetUserMe(): Promise<CustomerUserResponse> {
+export async function apiGetUserMe() {
     const response = await fetch(`${API_URL}/user/me`, {
         method: 'GET',
         headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
         mode,
         credentials,
     });
-    return await response.json();
+    return await response;
```
Comment (Diego-Romero) [inline src/redux/api.ts]: you don't need await here

**Pull request merged**